rs/tests/cross_chain/ic_xc_cketh_test.rs

```diff
@@ -0,0 +1,984 @@
+use hex_literal::hex;
+use ic_cketh_minter::endpoints::{CandidBlockTag, MinterInfo};
+use ic_cketh_minter::lifecycle::upgrade::UpgradeArg as MinterUpgradeArg;
+use ic_cketh_minter::lifecycle::{init::InitArg as MinterInitArgs, EthereumNetwork, MinterArg};
+use ic_cketh_minter::numeric::BlockNumber;
+use ic_consensus_system_test_utils::rw_message::install_nns_with_customizations_and_check_progress;
+use ic_consensus_threshold_sig_system_test_utils::{
+    enable_chain_key_signing, get_public_key_and_test_signature, make_key,
+};
+use ic_ethereum_types::Address;
+use ic_icrc1_ledger::{ArchiveOptions, FeatureFlags, InitArgsBuilder, LedgerArgument};
+use ic_ledger_suite_orchestrator::candid::{
+    AddErc20Arg, Erc20Contract, InitArg, LedgerInitArg, ManagedCanisterIds, OrchestratorArg,
+    UpgradeArg as LedgerSuiteOrchestratorUpgradeArg,
+};
+use ic_management_canister_types::{EcdsaKeyId, MasterPublicKeyId};
+use ic_nns_constants::{GOVERNANCE_CANISTER_ID, ROOT_CANISTER_ID};
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::driver::test_env_api::SubnetSnapshot;
+use ic_system_test_driver::driver::universal_vm::DeployedUniversalVm;
+use ic_system_test_driver::util::MessageCanister;
+use ic_system_test_driver::{
+    driver::{
+        group::SystemTestGroup,
+        ic::{InternetComputer, Subnet},
+        test_env::TestEnv,
+        test_env_api::{
+            get_dependency_path, HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer,
+            NnsCustomizations, SshSession,
+        },
+        universal_vm::{UniversalVm, UniversalVms},
+    },
+    systest,
+    util::{block_on, runtime_from_url},
+};
+use ic_types::Height;
+use icrc_ledger_types::icrc1::account::Account;
+use reqwest::Client;
+use slog::{info, Logger};
+use std::env;
+use std::future::Future;
+use std::time::Duration;
+
+const FOUNDRY_VM_NAME: &str = "foundry";
+const DOCKER_NETWORK_NAME: &str = "ethereum";
+const FOUNDRY_PORT: u16 = 8545;
+const ENCODED_PRINCIPAL: &str =
+    "0x1d9facb184cbe453de4841b6b9d9cc95bfc065344e485789b550544529020000";
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(setup_with_system_and_application_subnets)
+        .add_test(systest!(ic_xc_cketh_test))
+        .execute_from_args()?;
+    Ok(())
+}
+
+fn setup_with_system_and_application_subnets(env: TestEnv) {
+    setup_anvil(&env);
+    InternetComputer::new()
+        .add_subnet(Subnet::new(SubnetType::System).add_nodes(1))
+        .add_subnet(
+            Subnet::new(SubnetType::Application)
+                .add_nodes(1)
+                .with_dkg_interval_length(Height::from(10)),
+        )
+        .setup_and_start(&env)
+        .expect("Failed to setup IC under test");
+    install_nns_with_customizations_and_check_progress(
+        env.topology_snapshot(),
+        NnsCustomizations::default(),
+    );
+
+    env.topology_snapshot()
+        .subnets()
+        .for_each(|subnet| subnet.await_all_nodes_healthy().unwrap());
+}
```

Comment:
This isn't strictly necessary since `install_nns_with_customizations_and_check_progress` above already [waits for nodes to be healthy](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/consensus/utils/src/rw_message.rs?L275-L292).

Reply:
Thanks for the hint. Removed.